File: daily/2026/1411_Number_of_Ways_to_Paint_N×3_Grid.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int numOfWays(int n) {
        int mod = 1e9 + 7;
        long long ans = 0;
        vector<long long> dp(12, 1);
        for (int i = 2; i <= n; i++){
            vector<long long> temp(12, 0);
            temp[0] = (dp[4] + dp[5] + dp[7] + dp[8] + dp[9]) % mod;
            temp[1] = (dp[4] + dp[6] + dp[7] + dp[8]) % mod;
            temp[2] = (dp[4] + dp[5] + dp[8] + dp[9] + dp[11]) % mod;
            temp[3] = (dp[5] + dp[9] + dp[10] + dp[11]) % mod;
            temp[4] = (dp[0] + dp[1] + dp[2] + dp[10] + dp[11]) % mod;
            temp[5] = (dp[0] + dp[2] + dp[3] + dp[10]) % mod;
            temp[6] = (dp[1] + dp[8] + dp[9] + dp[11]) % mod;
            temp[7] = (dp[0] + dp[1] + dp[9] + dp[10] + dp[11]) % mod;
            temp[8] = (dp[0] + dp[1] + dp[2] + dp[6]) % mod;
            temp[9] = (dp[0] + dp[2] + dp[3] + dp[6] + dp[7]) % mod;
            temp[10] = (dp[3] + dp[4] + dp[5] + dp[7]) % mod;
            temp[11] =  (dp[2] + dp[3] + dp[4] + dp[6] + dp[7]) % mod;
            dp = temp;
        }
        for (long long i : dp) ans = (ans + i) % mod;
        return ans;
    }
};

int main(){
    int n = 1;
    // int n = 2;
    // int n = 10;
    // int n = 1000;
    // int n = 5000;
    Solution S;
    cout << S.numOfWays(n) << endl;
    return 0;
}

File: daily_python/1411_Number_of_Ways_to_Paint_N×3_Grid.py
import os
import sys
from typing import List

class Solution:
    def numOfWays(self, n: int) -> int:
        mod = 10 ** 9 + 7
        a = 6
        b = 6
        for _ in range(n - 1):
            next_a = (a * 3 + b * 2) % mod
            next_b = (a * 2 + b * 2) % mod
            a, b = next_a, next_b
        return (a + b) % mod
    
def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            n = int(lines[0].split('=')[1].strip()[:-1])
            dataset.append(n)
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for n in dataset:
        results.append(solution.numOfWays(n))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')